src/modules/luksopenswaphookcfg/LOSHJob.cpp

```diff
@@ -0,0 +1,187 @@
+        return Calamares::JobResult::internalError(
+            "LuksOpenSwapHook", tr( "No rootMountPoint is set." ), Calamares::JobResult::InvalidConfiguration );
+    }
+    if ( m_configFilePath.isEmpty() )
+    {
+        return Calamares::JobResult::internalError(
+            "LuksOpenSwapHook", tr( "No configFilePath is set." ), Calamares::JobResult::InvalidConfiguration );
+    }
+
+    QStringList contents = sys->readTargetFile( m_configFilePath );
+    if ( contents.isEmpty() )
+    {
+        contents << QStringLiteral( "# swap_device=" ) << QStringLiteral( "# crypt_swap_name=" )
+                 << QStringLiteral( "# keyfile_device=" ) << QStringLiteral( "# keyfile_filename=" )
+                 << QStringLiteral( "# keyfile_device_mount_options" );
+    }
+
+    write_openswap_conf( m_configFilePath, contents, LOSHInfo::fromGlobalStorage() );
+    return Calamares::JobResult::ok();
+}
+
+void
+LOSHJob::setConfigurationMap( const QVariantMap& configurationMap )
+{
+    m_configFilePath = CalamaresUtils::getString(
+        configurationMap, QStringLiteral( "configFilePath" ), QStringLiteral( "/etc/openswap.conf" ) );
+}
+
+STATICTEST void
+globalStoragePartitionInfo( Calamares::GlobalStorage* gs, LOSHInfo& info )
+{
+    if ( !gs )
+    {
+        return;
+    }
+    QVariantList l = gs->value( "partitions" ).toList();
+    if ( l.isEmpty() )
+    {
+        return;
+    }
+
+    for ( const auto& pv : l )
+    {
+        const QVariantMap partition = pv.toMap();
+        if ( !partition.isEmpty() )
+        {
+            QString mountPoint = partition.value( "mountPoint" ).toString();
+            QString fileSystem = partition.value( "fs" ).toString();
+            QString luksMapperName = partition.value( "luksMapperName" ).toString();
+            // if partition["fs"] == "linuxswap" and "luksMapperName" in partition:
+            if ( fileSystem == QStringLiteral( "linuxswap" ) && !luksMapperName.isEmpty() )
+            {
+                info.swap_outer_uuid = partition.value( "luksUuid" ).toString();
+                info.swap_mapper_name = luksMapperName;
+            }
+            else if ( mountPoint == QStringLiteral( "/" ) && !luksMapperName.isEmpty() )
+            {
+
+                info.mountable_keyfile_device = QStringLiteral( "/dev/mapper/" ) + luksMapperName;
+            }
+        }
+    }
+
+    if ( !info.mountable_keyfile_device.isEmpty() && !info.swap_outer_uuid.isEmpty() )
+    {
+        info.swap_device_path = QStringLiteral( "/dev/disk/by-uuid/" ) + info.swap_outer_uuid;
+    }
+
+    QString btrfsRootSubvolume = gs->value( "btrfsRootSubvolume" ).toString();
+    if ( !btrfsRootSubvolume.isEmpty() )
+    {
+        int slashCount = 0;
+        while ( slashCount < btrfsRootSubvolume.length() && btrfsRootSubvolume[ slashCount ] == '/' )
+        {
+            slashCount++;
+        }
+        if ( slashCount > 0 )
+        {
+            btrfsRootSubvolume.remove( 0, slashCount );
+        }
```

Comment:
Is there a reason this is being done in such a verbose way?

It seems like something like `btrfsRootSubvolume.remove( QRegExp( "^[/]*" ) );` would be more concise and readable.

Reply:
`"^/*"` without the brackets should be enough, and also, `QRegExp` is deprecated in favor of `QRegularExpression`.

Reply:
I'm not immediately convinced that "I know, I'll use a RegExp!" is an improvement in readability. I've taken a slightly different tack, since I'm somewhat surprised at QString lacking an `lstrip()` and `rstrip()` equivalent. I've bunged it into libcalamares as `removeLeading(btrfsRootSubvolume, '/')`.

Reply:
.. we can argue over the implementation there, but `removeLeading` is a more meaningful name than `"^/*"`